code/modules/research/experimentor/loot/_loot_definer.dm

```diff
@@ -0,0 +1,222 @@
+/////////////////////////////////////LOOT DEFINER/////////////////////////////////////
+// Holds all the messy stuff to do with name and type picking, loot lists etc.
+// define() returns a named unboxed item based on stats and loot lists.
+// The actual items and their purposes should be defined elsewhere,
+// though they can be initialized here as part of the loot picking.
+//
+// Expansion sections (Searchable):
+// O.   LEAVE THIS PART ALONE
+// I.   LOOT LIST
+// II.  LOOT DEFINITIONS
+// III. SECOND NAME
+// IV.  THIRD NAME
+// V.   DESCRIPTION
+
+
+#define RARITY_COMMON 0
+#define RARITY_UNCOMMON 1
+#define RARITY_RARE 2
+#define RARITY_VERYRARE 3
+
+//////////////////////// O. LEAVE THIS PART ALONE //////////////////////////
+// Unless you have a good idea of how this works.
+// If you're just adding new loot you shouldn't have to touch this.
+
+/obj/experimentor/loot_definer
+    name = "loot definer"
+    desc = "If you can see this, something is very, very wrong."
+    var/stability = 0
+    var/potency = 0
+    var/raritylevel = 0
+    var/box_type
+    var/item_category
+    var/item_type
+    var/obj/loot_item
+
+    // Set to FALSE during item definition to use the default name or description for the new object.
+    var/loot_named
+    var/loot_described
+    // Keywords used to choose names for the second and third parts of the generated name.
+    // When defining new keywords, try to use ones that fit in the auto-generated descriptions.
+    var/list/loot_keywords
+    // Unique Names - If instantiated in the definition, will use one of the contained names instead of generating the third name.
+    var/list/unique_names
+
+// Returns a finished item from the stats supplied to the experimentor.
+/obj/experimentor/loot_definer/proc/define(var/stability_in, var/potency_in, var/base_name, var/rare_level, var/itemcategory)
+	stability = stability_in
+	potency = potency_in
+	raritylevel = rare_level
+	box_type = base_name
+	item_category = itemcategory
+	loot_named = TRUE
+	loot_described = TRUE
+	loot_keywords = new/list
+	unique_names = null
+
+    // determine what specific type (itemtype) the item is based on.
+	item_type = findItemType()
+    // define the item to be granted as loot.
+	loot_item = definitionByType()
+	// name and describe the item unless the definition says not to.
+	if(loot_named && loot_item != null)
+		loot_item.name = box_type + " [generateSecondName()][generateThirdName()]"
+	if(loot_described && loot_item != null)
+		loot_item.desc = generateDescription()
+	return loot_item
+
+
+///////////////////I. LOOT LIST////////////////////
+// Defines the loot lists for the experimentor
+// by category then by rarity.
+
+/obj/experimentor/loot_definer/proc/findItemType()
+    switch (item_category)
```

Comment:
What's the purpose of this switch statement?

Reply:
Expandability. The system just has results of type "Device" for now, but eventually there will be more than one Type (loot list) to draw from.